### PR TITLE
test(cudf-polars): rationalize upstream Polars GPU gating

### PR DESCRIPTION
**Pull request by vyasr**
## Summary

- add per-test local/upstream coverage comparison and duplicate-test classification reports
- replace the positive upstream GPU allowlist with default-strict execution plus explicit CPU-fallback and expected-failure exceptions
- require listed fallback tests to record actual fallback, and run strict in-memory and SPMD upstream suites in CI

## Validation

- `python -m pytest python/cudf_polars/tests/testing/test_inject_gpu_engine.py -q`
  - `3 passed`
- Full upstream strict runs completed before the rebase:
  - in-memory: `34,782 passed, 101 skipped, 128 xfailed`
  - small-blocksize SPMD: `34,703 passed, 160 skipped, 148 xfailed`
- `git diff --check upstream/main...HEAD`
- `bash -n ci/run_cudf_polars_polars_tests.sh ci/run_cudf_polars_coverage_comparison.sh`

This is infrastructure and safety-gating only; it does not remove local cudf-polars tests.

### CODE REVIEW

File: ci/tests/test_analyze_cudf_polars_coverage.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+import importlib.util
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from types import ModuleType
+
+
+def _load_analyzer() -> ModuleType:
```
Comment (TomAugspurger) [inline ci/tests/test_analyze_cudf_polars_coverage.py]: This could hopefully be removed, if we require the tests to be invoked from the right path or modify PYTHONPATH in whatever script is launching it.
